packages/fxa-settings/src/pages/Signin/container.tsx

```diff
@@ -285,6 +292,35 @@ const SigninContainer = ({
         }
       );
 
+      // Check recovery key status if signin was successful, user is on sync Desktop
+      // and they didn't click "Do it later"; this affects navigation.
+      if (
+        'data' in result &&
+        result.data &&
+        options.service === 'sync'
+        // && localStorage.getItem('TODO')
+      ) {
+        // We must use auth-client here in case the user has 2FA or should be
+        // taken to signin_token_code, else GQL responds with 'Invalid token'
+
+        // todo: add try/catch
+        const { exists } = await authClient.recoveryKeyExists(
```

Comment:
I think Sync users always go to `signin_token_code` or the 2FA page if they have 2FA, but I’ve gotten myself into a state locally where I didn’t have to enter the code just the other day, so it was either this or pass the `authClient` function from the 3 container components down into the presentation component and into `handleNavigation` to call there. 

It was easiest to just call here but with this approach the session token isn't verified yet. Alternatively I could have returned this value at the `signIn` mutation which I started to do at first since I could check if `input.options.service === 'sync'`, but... it felt wrong returning a value like that in the mutation since I didn't see where we were doing something like that otherwise. (LMK if you disagree just so I know for the future.)